.github/workflows/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Tests
+
+on:
+  pull_request: ~
+  push:
+    branches: [ master ]
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-20.04"]
+        python-version: ["3.6", "3.11"]
+
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: ${{ matrix.python-version }}
+
+    name: Python ${{ matrix.python-version }} on OS ${{ matrix.os }}
+    steps:
+
+    - name: Acquire sources
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+        cache: 'pip'
+        cache-dependency-path: |
+          setup.cfg
+          setup.py
+
+    - name: Set up project
+      run: |
+        # Install package in editable mode.
+        pip install --use-pep517 --prefer-binary --editable=.[develop]
+
+        # Needs to be invoked manually upfront.
+        # https://github.com/python/mypy/issues/11103
+        mypy --install-types --non-interactive pytest_crate
+
+    - name: Run linters and software tests
+      run: |
+        pytest --cov pytest_crate -vvvs
+        coverage xml
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v3
+      with:
+        files: ./coverage.xml
+        flags: unittests
+        env_vars: OS,PYTHON
+        name: codecov-umbrella
+        fail_ci_if_error: false
```

Comment:
The repository would need to be added to https://codecov.io/ to make this work. In the meanwhile, `fail_ci_if_error: false` makes it so that it doesn't fail. Let me know if you have any objections about it.